partiql-parser/src/lalr/lexer.rs

```diff
@@ -570,29 +707,73 @@ mod tests {
 
     #[test]
     fn select() -> Result<(), LexicalError> {
-        let query = "SELECT g FROM data GROUP BY a";
-        let lexer = Lexer::new(query);
+        let query = "SELECT g\nFROM data\nGROUP BY a";
+        let mut offset_tracker = LineOffsetTracker::default();
+        let lexer = PartiqlLexer::new(query, &mut offset_tracker);
         let toks: Vec<_> = lexer.collect::<Result<_, _>>()?;
 
         assert_eq!(
             vec![
                 Token::Select,
                 Token::Identifier("g".to_owned()),
                 Token::From,
-                Token::Identifier("data".into()),
+                Token::Identifier("data".to_owned()),
+                Token::Group,
+                Token::By,
+                Token::Identifier("a".to_owned())
+            ],
+            toks.into_iter().map(|(_s, t, _e)| t).collect::<Vec<_>>()
+        );
+
+        assert_eq!(offset_tracker.num_lines(), 3);
+        assert_eq!(offset_tracker.at(query, 0), LineAndColumn::at(1, 1));
+        assert_eq!(offset_tracker.at(query, 1), LineAndColumn::at(1, 2));
+        assert_eq!(offset_tracker.at(query, 9), LineAndColumn::at(2, 1));
+        assert_eq!(offset_tracker.at(query, 19), LineAndColumn::at(3, 1));
```

Comment:
nit: could add a line for checking the end of the query. Something like

```
        assert_eq!(offset_tracker.at(query, 28), LineAndColumn::at(3, 10));
```

Reply:
Also, is there any way to easily check for panics in assertions (other than creating a test w/ the `should_panic` annotation)? I'm wondering if we should test the panic behavior of `at`:

https://github.com/partiql/partiql-lang-rust/blob/8b9ae1993671b94dcdb2b7a17d747a7c522b3bfc/partiql-parser/src/lalr/lexer.rs#L92-L96